docs/cli/app-types.md

```diff
@@ -0,0 +1,141 @@
+# .NET Core CLI types of portability of applications
+
+## Introduction
+There are several ways to think about the "types" of applications you can build. Usually, the types describe 
+a certain execution model or are based on what the application can "do"; examples of these are "console application", 
+"web application" etc. All of these types of applications (and more) can be created with .NET Core, since it is 
+a general purpose development platform.
+
+However, given its unique nature, .NET Core also has another axis on which to observe the type of the application 
```

Comment:
Should we clarify that the unique nature refers to its cross-platform capabilities? And instead of axis wouldn't angle work better here?